Migrate ci pipeline to github actions

carpasse opened the pull request:
Agreed to migrate to GH actions in https://github.com/expressjs/discussions/issues/229

The new ci action  and README.md changes are based on the respective files in [accepts](https://github.com/jshttp/accepts) repository

carpasse commented:
Action passing https://github.com/carpasse/resolve-path/actions/runs/9282562761

blakeembrey reviewed (commented):
This seems hard to maintain and not a direct port of the travis code. What's the logic behind changing mocha/nyc/instanbul in this PR? Why is the README changing to badge gen? 

It feels like it'd be simpler to run just the steps that you seem to trying to hack around on the versions supported. For example, no need for coverage on node 0.8.

Also, does the existing node action for installing versions not work here? 

Can we avoid doing this without changing the `package.json` file in the test process?

carpasse commented:
Hi @blakeembrey 

This isn't a direct port because the priority was maintaining consistency with all other GH actions in the express, pillarjs, and jshttp organizations. The change in Mocha was made to use a more modern version compatible with the latest Node.js. Additionally, replacing Istanbul with NYC aligns with how coverage is tracked in all other repositories.

The pipeline removes ESLint for Node.js versions 10 and below, which is consistent with previous actions. I can add it to the install step if you prefer.

carpasse commented:
May be worth mentioning that the travis build is not working at the moment
https://app.travis-ci.com/pillarjs/resolve-path

carpasse commented:
I've refactored the CI pipeline task, and I believe it's now more closely aligned with what Travis CI pipeline did.

The key differences from the Travis CI pipeline are:

- Added missing Node.js versions to the test matrix.
- Stopped linting for Node.js versions < 12 (previously < 6).
- Switched to NYC for test coverage collection, replacing Istanbul.

@blakeembrey  can you please take a look and let me know if there’s anything you think I should change?

UlisesGascon reviewed (commented):
Can we add support to Node@24 and merge it? :)
